src/app/plugins/scores.plugin.ts

```diff
@@ -62,81 +72,155 @@ export class ScoresPlugin extends Plugin {
 
         const visitorName = this._getTeamNameFromTeamData(visitorData);
         const homeName = this._getTeamNameFromTeamData(homeData);
 
+        //Check if game does not contain all search terms
+        if (
+          !(
+            this._strictlyContainsAllTokens(visitorName, teamArg) ||
+            this._strictlyContainsAllTokens(homeName, teamArg)
+          )
+        ) {
+          continue;
+        }
+
+        teamFound = true; //Designates as true if any game was found with matching search term
+
         const visitorScore = parseInt(visitorData[visitorData.length - 1]);
         const homeScore = parseInt(homeData[homeData.length - 1]);
 
-        //Boolean for if the searched team is the one winning
-        let winning = 0;
-        let opponentName = '';
-        let teamName = '';
-
         //If searched team is the visitor, else
-        if (this._containsAllTokens(visitorName, teamArg.split(' '))) {
-
-          if (visitorScore > homeScore) { winning = 1; }  //Winning
-          else if (visitorScore < homeScore) { winning = 0; } //Losing
-          else if (visitorScore == homeScore) { winning = 2; } //Tied
-
-          opponentName = homeName;
-          teamName = visitorName;
-        } else {
-
-          if (homeScore > visitorScore) { winning = 1; }  //Winning
-          else if (homeScore < visitorScore) { winning = 0; } //Losing
-          else if (homeScore == visitorScore) { winning = 2; } //Tied
-
-          opponentName = visitorName;
-          teamName = homeName;
-        }
+        const winning = this._strictlyContainsAllTokens(visitorName, teamArg) ?
+          this._evaluateScores(visitorScore, homeScore) : this._evaluateScores(homeScore, visitorScore);
+        const winningString = game.toLowerCase().includes('final') ? this._FINAL_LABELS[winning] : this._WINNING_LABELS[winning];
 
-        const winningString = this._WINNING_LABELS[winning];
+        const opponentName = this._strictlyContainsAllTokens(visitorName, teamArg) ? homeName : visitorName;
+        const teamName = this._strictlyContainsAllTokens(visitorName, teamArg) ? visitorName : homeName;
 
         const embed = new RichEmbed();
         embed.setTitle(visitorName + ' at ' + homeName);
         embed.setColor('#7289da');
-        embed.setDescription(
-          teamName + ' is currently ' + winningString + ' against ' + opponentName
-        );
+        if (!game.toLowerCase().includes('final')) {
+          embed.setDescription(
+            `${teamName} is currently ${winningString} against ${opponentName}`
+          );
+        } else {
+          embed.setDescription(
+            `${teamName} ${winningString} ${opponentName}`
+          );
+        }
         embed.setFooter(visitorScore + ' - ' + homeScore);
         embedBuffer.push(embed);
-
       }
 
       //Send embeds in buffer
-      embedBuffer.forEach(e => {
-        message.channel.send(e)
+      embedBuffer.forEach((e) => {
```

Comment:
why not `forEach(message.channel.send)`? take advantage of the fact you can pass the actual function, to be called by the implementation of `forEach`

Reply:
It throws an error when I do `forEach(message.channel.send)`